tests/e2e/test_catalog_api.py

```diff
@@ -0,0 +1,78 @@
+from typing import Generator
+
+import pytest
+import sqlalchemy
+from pydantic_core import to_jsonable_python
+from fastapi.testclient import TestClient
+
+from dor.adapters.catalog import Base, SqlalchemyCatalog, _custom_json_serializer
+from dor.config import config
+from dor.domain.models import Bin
+from dor.entrypoints.api.catalog import get_db_session
+from dor.entrypoints.api.main import app
+
+
+@pytest.fixture
+def db_session() -> Generator[sqlalchemy.orm.Session, None, None]:
+    engine_url = config.get_database_engine_url(test=True)
+    engine = sqlalchemy.create_engine(
+        engine_url, echo=True, json_serializer=_custom_json_serializer
+    )
+
+    Base.metadata.drop_all(engine)
+    Base.metadata.create_all(engine)
+
+    connection = engine.connect()
+    session = sqlalchemy.orm.Session(bind=connection)
+
+    yield session
+
+    session.close()
+    connection.close()
+
+
+@pytest.fixture
+def test_client(db_session) -> Generator[TestClient, None, None]:
+    def get_db_session_override():
+        return db_session
+
+    app.dependency_overrides[get_db_session] = get_db_session_override
+    test_client = TestClient(app)
+    yield test_client
+    app.dependency_overrides.clear()
```

Comment:
This is the solution to the database configuration problem, taken pretty verbatim from this section of the FastAPI/SQLModel docs: https://sqlmodel.tiangolo.com/tutorial/fastapi/tests/#client-fixture Basically, if you look at the regular `app` file, it uses the regular database (`dor_local` for local development), which we don't want to use for testing purposes. This code overrides the session dependency used by the current routes so that we use a session associated wtih our test database, `dor_test`.